src/nrepl/config.clje

```diff
@@ -0,0 +1,61 @@
+(ns nrepl.config
+  "Server configuration utilities.
+  Some server options can be configured via configuration
+  files (local or global).  This namespace provides
+  convenient API to work with them.
+
+  The config resolution algorithm is the following:
+  The global config file .nrepl/nrepl.edn is merged with
+  any local config file (.nrepl.edn) if present.
+  The values in the local config file take precedence."
+  {:author "Bozhidar Batsov"
+   :added  "0.5.0"}
+  (:require
+   [clojure.erlang.io :as io]
+   [clojure.edn :as edn]))
+
+(def ^:private home-dir
+  "The user's home directory."
+  (os/getenv #erl"HOME"))
+
+(def ^:private separator
+  "MB: File.separator
+  This should be taken from File.separator, but I couldn't find it."
+  "/")
+
+(def config-dir
+  "nREPL's configuration directory.
+  By default it's ~/.nrepl, but this can be overridden
+  with the NREPL_CONFIG_DIR env variable."
+  (or (os/getenv #erl"NREPL_CONFIG_DIR")
+      (str home-dir separator ".nrepl")))
+
+(def config-file
+  "nREPL's config file."
+  (str config-dir separator "nrepl.edn"))
+
+(defn- load-edn
+  "Load edn from an io/reader source (filename or io/resource)."
+  [source]
+  (with-open [r (io/reader source)]
+    (edn/read (erlang.io.PushbackReader. r))))
+
+(defn- load-config
+  "Load the configuration file identified by `filename`.
+  Return its contents as EDN if the file exists,
+  or an empty map otherwise."
+  [filename]
+  (let [file (io/file filename)]
+    (if (.exists file)
+      (load-edn file)
+      {})))
+
+(def config
```

Comment:
The reason why the compilation of this file is not working is because vars work slightly in a different way in Clojerl.

In Clojure JVM the root values of vars is calculated when the namespace is loaded. In Clojerl the value is calculated at compile time. The way to address this is to turn this var into a function and then either keep the resulting value somewhere when the application starts or (depending on how expensive the calculation is) keep calling the function everytime the value is needed.